sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/QueryInfo.java

```diff
@@ -160,6 +161,11 @@ public boolean hasGroupBy() {
         return groupByExpressions != null && !groupByExpressions.isEmpty();
     }
 
+    public boolean hasNonStreamingOrderBy() {
+        this.nonStreamingOrderBy = Boolean.TRUE.equals(super.getBoolean("hasNonStreamingOrderBy"));
```

Comment:
Do we want to pre-check if the payload has `hasNonStreamingOrderBy` ?

Reply:
As per the discussion with the BE team, the `QueryInfo` being passed by BE will have `hasNonStreamingOrderBy`. It would just be set to false for queries which are not non streaming order by queries.